### PR TITLE
[flashlight-sequence] New port

### PR DESCRIPTION
**Pull request by jacobkahn**
Adds a port for Flashlight Sequence (https://github.com/flashlight/sequence). Features include:
- `cuda` -- build CUDA kernels with the package 
- `openmp` -- build with OpenMP support

Tested ports on all platforms/features except for `x64-windows` (with CUDA), but we have a CI baseline for this -- https://app.circleci.com/pipelines/github/flashlight/sequence/65/workflows/3af72357-b9a7-4d66-aa58-5fa65892243b/jobs/526

---------

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/flashlight-sequence/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "flashlight-sequence",
+  "version": "0.0.1",
+  "description": "Flashlight Sequence is a library containing fast implementations of sequence loss algorithms.",
+  "homepage": "https://github.com/jacobkahn/sequence",
```
Comment (JonLiu1993) [inline ports/flashlight-sequence/vcpkg.json:5]:
```suggestion
  "homepage": "https://github.com/flashlight/sequence",
```

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

File: ports/flashlight-sequence/vcpkg.json (hunk 1 above)
Comment (jacobkahn) [reply ports/flashlight-sequence/vcpkg.json:5]: Thanks 😃

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Tested all features successfully in the following triplet:
- x64-windows
- x64-windows-static
Tested usage successfully by `flashlight-sequence:x64-windows`:
````
Total install time: 39 s
flashlight-sequence provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(flashlight-sequence CONFIG REQUIRED)
    target_link_libraries(main PRIVATE flashlight::flashlight-sequence)
````

**Pull request merged**